crates/rethnet_evm_napi/src/transaction/signed/eip1559.rs

```diff
@@ -30,21 +25,9 @@ pub struct EIP1559SignedTransaction {
 impl EIP1559SignedTransaction {
     /// Constructs a [`EIP1559SignedTransaction`] instance.
     pub fn new(
-        env: &Env,
         transaction: &rethnet_eth::transaction::EIP1559SignedTransaction,
     ) -> napi::Result<Self> {
-        let input = transaction.input.clone();
-        let input = unsafe {
-            env.create_buffer_with_borrowed_data(
-                input.as_ptr(),
-                input.len(),
-                input,
-                |input: Bytes, _env| {
-                    mem::drop(input);
-                },
-            )
-        }
-        .map(JsBufferValue::into_raw)?;
```

Comment:
This code was intentional. Using a borrow, it avoids a copy of the memory, since `Bytes` is able to share memory

Reply:
I had to switch the `JsBuffer` to `Buffer` to make the `rethnet_evm_napi::SignedTransaction` type `Sync` so that the async `rethnet_evm_napi::debug_trace_transaction` function can take `rethnet_evm_napi::SignedTransaction` as argument. 

Alternative would be to have two types: `rethnet_evm_napi::SignedTransactionToJs` with `JsBuffer` fields and `rethnet_evm_napi::SignedTransactionFromJs` with `Buffer` fields. Wdyt?

Reply:
Is the problem resolved by using a `PendingTransaction`?

Reply:
Yes, resolved by using `PendingTransaction`.

The original code looked like this:

```rust
#[napi]
pub async fn foo(buffer: napi::JsBuffer) {}
```

And it failed with this error:

```
error: future cannot be sent between threads safely
  --> crates/rethnet_evm_napi/src/debug_trace.rs:14:1
   |
14 | #[napi]
   | ^^^^^^^ future created by async block is not `Send`
   |
   = help: within `[async block@crates/rethnet_evm_napi/src/debug_trace.rs:14:1: 14:8]`, the trait `Send` is not implemented for `*mut napi_env__`
note: captured value is not `Send`
  --> crates/rethnet_evm_napi/src/debug_trace.rs:14:1
   |
14 | #[napi]
   | ^^^^^^^ has type `JsBuffer` which is not `Send`
note: required by a bound in `execute_tokio_future`
  --> /Users/abiro/.cargo/registry/src/index.crates.io-6f17d22bba15001f/napi-2.13.3/src/tokio_runtime.rs:89:18
   |
89 |   Fut: 'static + Send + Future<Output = Result<Data>>,
   |                  ^^^^ required by this bound in `execute_tokio_future`
   = note: this error originates in the attribute macro `napi` (in Nightly builds, run with -Z macro-backtrace for more info)
```